### PR TITLE
New version required and modifications for next step with Home Assistant

### PR DESCRIPTION
**Pull request by gjohansson-ST**
Hi

Github version is not the same as published on pypi which we need to fix to take next steps getting the integration well functioning in HA.
Also backoff can not be an explicit version so modified == to >= to accomodate newer version.

Last but not least I added a few functions to get the full device and mode list out so it's easier to get the full json object back for various states (at least right now during dev).

Hope it can be approved and then uploaded to pypi shortly for version 3.3.

### CODE REVIEW

**Comment by gjohansson-ST** [comment]
@domwillcode I hope you can approve this and get it merged quite soon.
Also need to bump the version on pypi to 3.3 to take next steps with the HA integration.
Se PR here with HA which I am stuck with now https://github.com/home-assistant/core/pull/50613

**Comment by domwillcode** [comment]
Cheers for this! No worries, on mobile at the mo but will push a new version tomorrow.

**Comment by gjohansson-ST** [comment]
> Cheers for this! No worries, on mobile at the mo but will push a new version tomorrow.

Waiting is not one of my strengths ;)

btw; is this package used elsewhere given how it's designed? Seems not too designed for usage with HA.

File: yalesmartalarmclient/client.py [hunk 1]
```diff
@@ -47,7 +51,48 @@ def __init__(self, username, password, area_id=1):
         self.area_id = area_id
         self.lock_api: YaleDoorManAPI = YaleDoorManAPI(auth=self.auth)
 
-    # Kept for backwards compatibility.
+    # Included to get full visibility from API for local testing, use with print()
+    def get_all(self):
```
Comment (domwillcode) [inline yalesmartalarmclient/client.py:55]: A get on this one feels misleading when returning a string vs the raw dict but it won't affect HA so in the interest of getting this in I'll hit go.

**Review by domwillcode — APPROVED** [review]

**Comment by domwillcode** [comment]
> > Cheers for this! No worries, on mobile at the mo but will push a new version tomorrow.
> 
> Waiting is not one of my strengths ;)
> 
> btw; is this package used elsewhere given how it's designed? Seems not too designed for usage with HA.

Not to my knowledge. It's 'designed' this way to slot into what HA looked like 2-3 years ago. Feel free to go crazy :)

File: yalesmartalarmclient/client.py (hunk 1 above)
Comment (gjohansson-ST) [reply yalesmartalarmclient/client.py:55]: > A get on this one feels misleading when returning a string vs the raw dict but it won't affect HA so in the interest of getting this in I'll hit go.

Thanks. It's only there to get a full output when running it manually so the guys with all the equipment could easily just send it to me as a full textblock. We should remove this function in the next version.

**Comment by gjohansson-ST** [comment]
> > > Cheers for this! No worries, on mobile at the mo but will push a new version tomorrow.
> > 
> > 
> > Waiting is not one of my strengths ;)
> > btw; is this package used elsewhere given how it's designed? Seems not too designed for usage with HA.
> 
> Not to my knowledge. It's 'designed' this way to slot into what HA looked like 2-3 years ago. Feel free to go crazy :)

Then I will do so 😄 
for now mainly interested to get v3.3 to pypi so I can continue in HA dev.

Thanks

**Comment by domwillcode** [comment]
Cheers. I will push to pypi in a few hours, please hold! ☺️

**Comment by domwillcode** [comment]
https://pypi.org/project/yalesmartalarmclient/0.3.3/ Job done!

**Pull request merged**